lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -1150,6 +1191,180 @@ void SESERegionBuilder::ensureSingleExitFromLoops() {
   }
 }
 
+void SingleExitLoopTransformer::unrollLoopBodyOnce() {
+  // Consider the following example:
+  //
+  //   do {
+  //     i += 1
+  //     if (...) break;
+  //     i += 1
+  //   } while(...)
+  //   return i
+  //
+  // The SIL CFG is shown below:
+  //
+  //   preheader:
+  //     i0 = 0
+  //     br header(i0)
+  //
+  //   header(i1):
+  //     i2 = i1 + 1
+  //     cond ??, break, body
+  //
+  //   break:
+  //     br exit(i2)
+  //
+  //   body:
+  //     i3 = i2 + 1
+  //     cond ??, header(i3), exit(i3)
+  //
+  //   exit(i4):
+  //     return i4
+  //
+  // Note that the dataflow between iterations of the loop is captured by
+  // argument `i1` of the header. If we need to exit from the header of the
+  // loop, we will need to "freeze" the state at the current exit points and
+  // propagate it to the new header. The canonicalization pass does that by
+  // adding an argument to the header and passing it from the exit points. See
```

Comment:
what is "from the exit points"? i believe you are not talking about the "exit" block below.

Reply:
I meant exit points from the loop. Clarified the comment.